ddtrace/span.py

```diff
@@ -165,17 +165,9 @@ def finish(self, finish_time=None):
             self.duration_ns = ft - (self.start_ns or ft)
 
         if self._context:
-            try:
-                self._context.close_span(self)
-            except Exception:
-                log.exception("error recording finished trace")
-            else:
-                # if a tracer is available to process the current context
-                if self.tracer:
-                    try:
-                        self.tracer.record(self._context)
-                    except Exception:
-                        log.exception("error recording finished trace")
+            trace, sampled = self._context.close_span(self)
+            if self.tracer and trace and sampled:
+                self.tracer.write(trace)
```

Comment:
this is an aside, nothing to act on now.

I wonder if we should pass `sampled` through to `self.tracer.write` for logging.

since right now we would skip over debug logging when a trace is not sampled, but would be really nice to have a debug message saying a trace was skipped (metrics is probably better, but thought still stands?)

Reply:
Wouldn't it make sense to log where the sampling decision is actually made?

Reply:
sure, was just noticing that with this codepath we are likely missing that debug logging for something that is not sampled. idc about where/when/etc. something to think about.